src/Arriba/Arriba/Diagnostics/Tracing/ArribaEventSource.cs

```diff
@@ -63,9 +64,20 @@ public interface IArribaEvents : IEventSource
 
     public sealed class ArribaEventSource : EventSource, IArribaEvents
     {
+
+        public delegate  void TimedFunction();
+        
         public static IArribaEvents Log { get; }
         public static IArribaServiceEvents Service { get; }
 
+
+        public static Task<TimeSpan> Time(TimedFunction timedFunction)
```

Comment:
This should be a Func<Task> that way it is async and can be awaited. The current coke will produce unexpected results and a crash may result in the entire app be being torn down as the exception cannot be appropriately handled

Reply:
Changed for an Action